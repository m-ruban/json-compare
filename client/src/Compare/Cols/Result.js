import React from 'react'
import {Col} from 'reactstrap'
import Prop from '../../JSON/Prop'

const Result = ({obj, showAlerts, showRequired}) => (
  <Col xs="7">
    <div className="json result without-panel">
      <Prop
        value={obj}
        showAlerts={showAlerts}
        showRequired={showRequired}
        lvl={1}
      />
    </div>
  </Col>
)

export default Result