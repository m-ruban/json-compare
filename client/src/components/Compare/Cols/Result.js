import React from 'react';
import PropTypes from 'prop-types';
import { Col } from 'reactstrap';
import Prop from 'components/JSON/Prop';

const Result = ({ obj, showAlerts, showRequired }) => (
    <Col xs="7">
        <div className="json result without-panel">
            <Prop value={obj} showAlerts={showAlerts} showRequired={showRequired} lvl={1} />
        </div>
    </Col>
);

Result.propTypes = {
    obj: PropTypes.object,
    showAlerts: PropTypes.bool,
    showRequired: PropTypes.bool,
};

Result.defaultProps = {
    showAlerts: false,
    showRequired: false,
};

export default Result;
