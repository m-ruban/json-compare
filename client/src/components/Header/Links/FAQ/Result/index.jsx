import React from 'react';
import { Badge } from 'reactstrap';
import trl from 'modules/translation';

const Result = () => (
    <div>
        <ol>
            <li>
                <Badge color="req" pill>
                    !
                </Badge>
                {trl('FAQ.ComparisonResult.required')}
            </li>
            <li>
                <Badge color="diff" pill>
                    !
                </Badge>
                {trl('FAQ.ComparisonResult.difference')}
            </li>
        </ol>
    </div>
);

export default Result;
