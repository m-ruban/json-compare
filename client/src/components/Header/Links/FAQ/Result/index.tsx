import React from 'react';

import Mark from 'components/Mark';
import trl from 'modules/translation';

const Result = () => (
    <div>
        <ol>
            <li>
                <Mark type="req" text="!" pill />
                {trl('FAQ.ComparisonResult.required')}
            </li>
            <li>
                <Mark type="diff" text="!" pill />
                {trl('FAQ.ComparisonResult.difference')}
            </li>
        </ol>
    </div>
);

export default Result;
