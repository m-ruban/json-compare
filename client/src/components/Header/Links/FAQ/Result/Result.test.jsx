import React from 'react';
import { render } from 'enzyme';
import Result from 'components/Header/Links/FAQ/Result';

describe('<Result />', () => {
    it('Renders correctly', () => {
        const output = render(<Result />);
        expect(output).toMatchSnapshot();
    });
});
