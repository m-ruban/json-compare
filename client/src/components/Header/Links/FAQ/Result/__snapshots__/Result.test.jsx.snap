// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Result /> Renders correctly 1`] = `
<div>
  <ol>
    <li>
      <span
        class="json-mark json-mark_req badge badge-json badge-pill"
      >
        !
      </span>
       - required key
    </li>
    <li>
      <span
        class="json-mark json-mark_diff badge badge-json badge-pill"
      >
        !
      </span>
       - the key exists, but it's value doesn't match the required one
    </li>
  </ol>
</div>
`;
